resource-timing/memory-cache-behavior.tentative.html

```diff
@@ -0,0 +1,172 @@
+        const p = document.createElement('p');
+        p.innerText = 'abc';
+        p.style.fontFamily = 'MyFont';
+        document.body.appendChild(p);
+        await document.fonts.ready;
+        return () => {
+            p.remove();
+            style.remove();
+        };
+    },
+    shape: (href, attr) => {
+        const div = document.createElement('div');
+        div.style.shapeOutside = `url(${href})`;
+        document.body.appendChild(div);
+        return () => div.remove();
+    },
+    backgroundImage: (href, attr) => {
+        const div = document.createElement('div');
+        div.style.background = `url(${href})`;
+        document.body.appendChild(div);
+        return () => div.remove();
+    },
+    script: async (href, attr) => {
+        const script = document.createElement('script');
+        if (attr.crossOrigin)
+            script.setAttribute('crossorigin', attr.crossOrigin);
+        script.src = href;
+        document.body.appendChild(script);
+        await new Promise(resolve => { script.onload = resolve });
+        return () => script.remove();
+    },
+    module: async (href, attr) => {
+        const script = document.createElement('script');
+        script.type = 'module';
+        const cleanup = () => script.remove();
+        if (attr.crossOrigin)
+            script.setAttribute('crossorigin', attr.crossOrigin);
+        script.src = href;
+        document.body.appendChild(script);
+        await new Promise(resolve => { script.onload = resolve });
+        return cleanup;
+    },
+    style: async (href, attr) => {
+        const style = document.createElement('link');
+        style.rel = 'stylesheet';
+        const cleanup = () => style.remove();
+        style.href = href;
+        if (attr.crossOrigin)
+            style.setAttribute('crossorigin', attr.crossOrigin);
+        document.head.appendChild(style);
+        await new Promise(resolve => {
+            style.addEventListener('load', resolve);
+            style.addEventListener('error', resolve);
+        });
+
+        return cleanup;
+    }
+}
+
+function memory_cache_reuse_test(type, cacheType, deadOrAlive, as, url, cfg1, cfg2) {
+    const expected = (cfg1 === cfg2 && cacheType !== 'no-store') ? "reuse" : "discard";
+    const key = token();
+    let href = getAbsoluteURL(`${
+        (configs[cfg2].crossOrigin ? HTTPS_REMOTE_ORIGIN : '') + url
+    }&${token()}`);
+    
+    if (cacheType !== 'none') 
+        href += `&cache=${encodeURIComponent(cacheType)}`
+    promise_test(async t => {
+        const remove = await loaders[as](href, configs[cfg1].attributes);
+        if (deadOrAlive === 'dead')
+            await remove();
+        else
+            t.add_cleanup(remove);
+//        await new Promise(resolve => t.step_timeout(resolve, 100));
+        const remove2 = await loaders[as](href, configs[cfg2].attributes);
+        t.add_cleanup(remove2);
+        await new Promise(resolve => t.step_timeout(resolve, 100));
```

Comment:
To reduce the chances of flakiness and the time it takes for these tests to run, it'd be better if for "discard" tests, we'd just use PerfObserver and not a timeout. For "reuse" tests, we don't have a choice, but for that case, I'm not sure if 100ms is enough. Maybe for that case, we can add another unrelated request, and check the results once that request returns? That way, we'd align the "timeout" with the server's responsiveness.